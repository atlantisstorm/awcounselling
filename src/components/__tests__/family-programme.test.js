import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import FamilyProgramme from '../family-programme';

describe('FamilyProgramme component', () => {
  it('should match snapshot', () => {
    const onClick = jest.fn();
    const { asFragment } = render(
      <FamilyProgramme onClick={ onClick } />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should contain contact opton link', () => {
    const onClick = jest.fn();
    const { getByTestId } = render(
      <FamilyProgramme onClick={ onClick } />
    );

    const link = getByTestId("family-programme--contact");
    fireEvent.click(link);
    expect(onClick).toHaveBeenCalledTimes(1);

    const dataName = link.getAttribute("data-name");
    expect(dataName).toBe("contact");
  });
});