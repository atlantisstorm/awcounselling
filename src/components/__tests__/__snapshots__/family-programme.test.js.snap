// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FamilyProgramme component should match snapshot 1`] = `
<DocumentFragment>
  <h5>
    One Day Family Programme
  </h5>
  <p>
    I run a one-day education/family programme to help create awareness about addiction and its effects on the addicted person and their family.
  </p>
  <p>
    Who is the programme aimed at? :
  </p>
  <ul>
    <li>
      Family and friends of people who use and abuse alcohol, drugs and/or gambling.
    </li>
    <li>
      People who are worried about their own use or abuse of alcohol, drugs or gambling.
    </li>
    <li>
      People who work or intend to work with people who suffer from addictions.
    </li>
    <li>
      This workshop has received A.C.I. approval for 6 C.P.D. points.
    </li>
  </ul>
  <p>
    Topics discussed :
  </p>
  <ul>
    <li>
      Chemical dependency
    </li>
    <li>
      Compulsive gambling
    </li>
    <li>
      Understanding the 12 steps of recovery
    </li>
    <li>
      Effects on the family and the different family roles
    </li>
    <li>
      Co-dependency
    </li>
    <li>
      How we enable addiction and how to detach from it
    </li>
    <li>
      Self-care
    </li>
    <li>
      Guest speakers from 12 step recovery group
    </li>
    <li>
      Questions and answers
    </li>
  </ul>
  <p>
    If you wish to find out when and where the next programme will be held or would like information on how to have me present the programme for your organisation, community or group then please contact me - click 
    <a
      data-name="contact"
      data-testid="family-programme--contact"
      href="/"
    >
      here
    </a>
     for contact details.
  </p>
</DocumentFragment>
`;
